extensions/android/BUILD.gn

```diff
@@ -0,0 +1,22 @@
+import("//build/config/android/rules.gni")
+
+android_library("xwalk_core_extensions_java") {
+  deps = [
+    "//base:base_java",
+    "//content/public/android:content_java",
+  ]
+  java_files = [
+    "java/src/org/xwalk/core/internal/extensions/XWalkExtensionAndroid.java",
+    "java/src/org/xwalk/core/internal/extensions/XWalkNativeExtensionLoaderAndroid.java",
+  ]
+}
+
+generate_jni("xwalk_core_extensions_native_jni") {
+  jni_package = rebase_path(
```

Comment:
There's no need to generate the headers in the same location as gyp, this is all internal to the build. As long as the right `-I<path>` is being passed (which `generate_jni` takes care of), there's no problem if the path is different.